src/test/starknet.t.sol

```diff
@@ -154,4 +161,48 @@ contract StarknetTests is DssSpellTestBase, ConfigStarknet {
 
         assertTrue(core.isNotFinalized());
     }
+
+    function _checkStarknetMessage(bytes32 _spell) internal {
+        StarknetCoreLike core = StarknetCoreLike(addr.addr("STARKNET_CORE"));
+
+        if (_spell != 0) {
+
+            // Nonce increments each message, back up one
+            uint256 _nonce = core.l1ToL2MessageNonce() - 1;
+
+            // Payload must be array
+            uint256[] memory _payload = new uint256[](1);
+            _payload[0] = uint256(_spell);
+
+            // Hardcoded in L1 gov relay, not public
+            uint256 RELAY_SELECTOR = 300224956480472355485152391090755024345070441743081995053718200325371913697;
+
+            // Hash of message created by Starknet Core
+            bytes32 _message = _getL1ToL2MsgHash(addr.addr("STARKNET_GOV_RELAY"), starknetValues.l2_gov_relay, RELAY_SELECTOR, _payload, _nonce);
+
+            // Assert message is scheduled, core returns 0 if not in message array
+            assertTrue(core.l1ToL2Messages(_message) > 0, "StarknetTest/SpellNotQueued");
```

Comment:
StarknetCore contract emits `LogMessageToL2`. It is better object to test against than the content of `l1Tol2Messages` which should be considered an implementation detail.
The way messages are hashed might change in the future.

Reply:
This is a good idea. Can use `vm.expectEmit` for that.

Reply:
`emit LogMessageToL2(msg.sender, toAddress, selector, payload, nonce, msg.value);`

This value is returned piecemeal and will still need to be re-hashed in the spell to reconstruct the message in the same manner as above. I don't think it gets us around any implementation-specific details, so the tests will need to be updated if this implementation changes the hash parameters in either case.

Reply:
The fully hashed message is returned from the 

`StarkNetLike(starkNet).sendMessageToL2{value: msg.value}(l2GovernanceRelay, RELAY_SELECTOR, payload);` 

call. If anything, the next version of the Gov Relay should probably include an event that captures that message and emits the return from it.

Reply:
Technically, Starknet sequencer initates l2 transactions based on `LogMessageToL2` events. `l1Tol2Messages` prevents it from including spoofed transactions that were not initiated from l1.